### PR TITLE
travelmate: add support for optional args in auto-login script

### PR DESCRIPTION
**Pull request by onjen**
Maintainer: @dibdot 
Compile tested: /
Run tested: QCA9533, GL.iNet GL-AR300M (NAND), OpenWrt SNAPSHOT r10759+12-b133e466b0 / LuCI Master (git-19.261.22678-982b08f) , Tested editing existing connections and adding new ones with a login script with additional arguments

Description: This can be helpful for example in hotels where you need to
enter a new user/password combination every week. Another PR will be opened in luci.

### CODE REVIEW

File: net/travelmate/files/generic-user-pass.login [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+response=$("${cmd}" $url -d "password=$2&pwd=$2&username=$1" \
+	--header "Content-Type:application/x-www-form-urlencoded" -s)
+
+if echo $response | grep -q "Thank you!";
+then
+	exit 0
+else
+	exit 1
```
Comment (dibdot) [inline net/travelmate/files/generic-user-pass.login]: please use only unique exit codes, e.g. exit with "2" on a failed login attempt

File: net/travelmate/files/travelmate.sh [hunk 2]
```diff
@@ -291,11 +291,19 @@ f_check()
 								trm_connection="${result:-"-"}/${trm_ifquality}"
 								f_jsnup
 								login_command="$(uci_get "travelmate" "${uci_essid}${uci_bssid}" "command")"
+								login_command_args="$(uci_get "travelmate" "${uci_essid}${uci_bssid}" "command_args")"
 								if [ -x "${login_command}" ]
 								then
-									"${login_command}" >/dev/null 2>&1
-									rc=${?}
-									f_log "info" "captive portal login '${login_command:0:40}' for '${cp_domain}' has been executed with rc '${rc}'"
+									if [ "${login_command_args}" ]
```
Comment (dibdot) [inline net/travelmate/files/travelmate.sh]: We don't need the if /else clause at all, it makes no harm if "login_command_args" is empty.

File: net/travelmate/files/travelmate.sh [hunk 3]
```diff
@@ -291,11 +291,19 @@ f_check()
 								trm_connection="${result:-"-"}/${trm_ifquality}"
 								f_jsnup
 								login_command="$(uci_get "travelmate" "${uci_essid}${uci_bssid}" "command")"
+								login_command_args="$(uci_get "travelmate" "${uci_essid}${uci_bssid}" "command_args")"
```
Comment (dibdot) [inline net/travelmate/files/travelmate.sh]: Please put this additional uci_get behind the first "login_command" test

**Comment by onjen** [comment]
First of, thanks to you for the awesome package! It makes the life as a road warrior so much easier :)

The latest push adresses the comments in this PR. I will test it again next week and report back.

**Comment by dibdot** [comment]
Many thanks, looks good now. I'll test your changes (including your LuCI PR) next week as well.

**Pull request merged**